core/server/services/auth/session/store.js

```diff
@@ -0,0 +1,44 @@
+const {Store} = require('express-session');
+const common = require('../../../lib/common');
+
+module.exports = class SessionStore extends Store {
+    constructor(SessionModel) {
+        super();
+        this.SessionModel = SessionModel;
+    }
+
+    destroy(sid, callback) {
+        this.SessionModel
+            .destroy({session_id: sid})
+            .then(() => {
+                callback(null);
+            })
+            .catch(callback);
+    }
+
+    get(sid, callback) {
+        this.SessionModel
+            .findOne({session_id: sid})
+            .then((model) => {
+                if (!model) {
+                    return callback(null, null);
+                }
+                callback(null, model.get('session_data'));
+            })
+            .catch(callback);
+    }
+
+    set(sid, sessionData, callback) {
+        if (!sessionData.user_id) {
+            return callback(new common.errors.InternalServerError({
```

Comment:
Is that an internal server error? 🤔 Looks like validation or permission.

Reply:
Because we create the session, I see it as an internal server error, if it was a 4xx error it would imply the user has done something wrong. But there is no way for a user to force this state, so it's a problem our end?

Reply:
Okay :)